bokeh/core/has_props.py

```diff
@@ -607,8 +607,17 @@ def apply_theme(self, property_values):
 
         # Emit any change notifications that result
         for k, v in old_values.items():
-            prop = self.lookup(k)
-            prop.trigger_if_changed(self, v)
+            descriptor = self.lookup(k)
+            if isinstance(v, PropertyValueContainer) and v._unmodified_default_value:
+                if k in self._property_values[k]:
+                    self._property_values[k]._unregister_owner(self, descriptor)
+                if k in property_values:
+                    self._property_values[k] = descriptor.property.prepare_value(self.__class__, k, property_values[k])
+                else:
+                    self._property_values[k] = descriptor.class_default(self.__class__)
+                self._property_values[k]._unmodified_default_value = True
+                self._property_values[k]._register_owner(self, descriptor)
+            descriptor.trigger_if_changed(self, v)
```

Comment:
container properties get stored in `_property_values` even on a read-only "get", that happens in this line:

https://github.com/bokeh/bokeh/blob/master/bokeh/core/has_props.py#L601

But then that means when `trigger_if_changed` is called later, here:

https://github.com/bokeh/bokeh/blob/master/bokeh/core/has_props.py#L611

nothing happens, because the "old" value passed in is the same one the "get" above put in `_property_values`